src/lib/components/UrlChooser/index.js

```diff
@@ -0,0 +1,158 @@
+import {html} from "lit-element";
+import {BaseElement} from "../BaseElement";
+
+/**
+ * @fileoverview Displays the primary URL chooser for Lighthouse.
+ */
+
+/* eslint-disable require-jsdoc */
+class UrlChooser extends BaseElement {
+  static get properties() {
+    return {
+      url: {type: String},
+      active: {type: Boolean},
+      switching: {type: Boolean, reflect: true}, // FIXME: internal but must reflect as [switching]
+    };
+  }
+
+  constructor() {
+    super();
+    this.url = null; // when signed out or waiting for Firestore, this is null
+    this.switching = true; // controls whether the user is editing the URL
+    this.active = false; // disables buttons (because Lighthouse is active)
+
+    // non-properties (stolen DOM nodes)
+    this._urlInput = null;
+    this._runLighthouseButton = null;
+  }
+
+  render() {
+    return html`
+      <div class="report_header_enterurl">
+        <div class="lh-enterurl lh-enterurl--selected">${this.url}</div>
+        <div class="lh-enterurl lh-enterurl--switch">
+          <input
+            type="url"
+            class="lh-input"
+            placeholder="Enter a web page URL"
+            pattern="https?://.*"
+            minlength="7"
+            @keyup="${this.onUrlKeyup}"
+          />
+          <button
+            class="lh-enterurl__close"
+            aria-label="Remove URL"
+            @click=${this.clearInput}
+          ></button>
+        </div>
+        <div class="lh-controls">
+          <button
+            ?disabled=${this.active}
+            class="w-button w-button--secondary"
+            @click=${this.switchUrl}
+          >
+            Switch URL
+          </button>
+          <button
+            ?disabled=${this.active}
+            class="w-button w-button--primary"
+            id="run-lh-button"
+            @click=${this.requestAudit}
+          >
+            Run Audit
+          </button>
+        </div>
+      </div>
+    `;
+  }
+
+  updated(changedProperties) {
+    this._urlInput = this.renderRoot.querySelector('input[type="url"]');
+    this._runLighthouseButton = this.renderRoot.querySelector("#run-lh-button");
+
+    if (!changedProperties.has("url")) {
```

Comment:
As discussed in the other PR, property change handlers are pure functions. I've made this a getter/setter for `url`.

Reply:
Actually I know I resolved this but it doesn't work.

We need to be able to set the value on `<input />` and we don't have it until after a first update. We don't want to have a property for that as it might blindly overwrite what the user is currently typing.

I can't think of a better solution than this?